lib/Teams.js

```diff
@@ -11,14 +11,17 @@ function TeamsBot(configuration) {
     var controller = Botkit(configuration || {});
 
     controller.api = TeamsAPI(configuration || {});
-
-    controller.api.getToken(function(err) {
+    function tokenHandler(err) {
         if (err) {
             // this is a fatal error - could not create a Teams API client
             throw new Error(err);
         }
-    });
-
+        //use the expires_in value, convert to ms and remove 10 mins in ms
+        let tokenExpiresIn = 1000 * 60 * (configuration.token_expires_in || 60);
```

Comment:
expires_in unit is seconds, for example the value "3600" will denote that token will expire in one hour.
You may have to change the above line and update tests 
` let tokenExpiresIn = 1000 * (configuration.token_expires_in || 3600) `